Enable GZIP compression for keycloak-gatekeeper

13rentgen opened the pull request:
* Implement GZIP compression for all incoming requests

abstractj commented:
@stianst this PR introduces another extra dependency only to achieve gzip compression and I'm really not sure if this is something we would like to introduce on Louketo. Maybe, might work to introduce it as an option, instead of making it the default. For example: `--enable-compression`.

Anyways, if we agree to move forward with these changes, we need tests for it.

stianst commented:
I'd say GZIP compression is something we should support right, so picking the correct library for the task seems like the way to go.

abstractj commented:
> I'd say GZIP compression is something we should support right, so picking the correct library for the task seems like the way to go.

+1

13rentgen commented:
> Could you please refactor to https://golang.org/pkg/compress/gzip/ ?

Refactored. Disable compression by default, will be enabled on 'enable-compression'. Added tests.

abstractj commented:
Thank you @13rentgen!
